protocol/synthetix/contracts/storage/Pool.sol

```diff
@@ -507,4 +528,25 @@ library Pool {
             );
         }
     }
+
+    function checkPoolCollateralLimit(
+        Data storage self,
+        address collateralType,
+        uint256 collateralAmountD18
+    ) internal view {
+        uint256 maxDeposit = self.collateralConfigurations[collateralType].maxDepositD18;
+
+        if (
+            (self.collateralDisabledByDefault && maxDeposit == 0) ||
+            (maxDeposit > 0 &&
+                self.vaults[collateralType].currentCollateral() + collateralAmountD18 > maxDeposit)
```

Comment:
Could we simplify this a bit, maybe event revert with some more specific errors? 
Should we revert if 
`collateralDisabledByDefault = false` and `maxDeposit = 0`?